base/src/com/google/idea/blaze/base/sync/ProjectUpdateSyncTask.java

```diff
@@ -285,6 +286,10 @@ private void run(BlazeContext context) throws SyncCanceledException, SyncFailedE
 
   private static void refreshVirtualFileSystem(
       BlazeContext context, Project project, BlazeProjectData blazeProjectData) {
```

Comment:
This is a private method which is called only once. I would rather perform this check before this method, not inside it.